src/r_data.c

```diff
@@ -1202,6 +1205,89 @@ boolean R_IsPatchLump (const int lump)
   return result;
 }
 
+typedef struct
+{
+  short width, height;  // bounding box size 
+  short leftoffset;     // pixels to the left of origin 
+  short topoffset;      // pixels below the origin 
+  short columnofs[8];   // only [width] used
+} patch_beta_t;
+
+patch_t *R_PatchByNum(int num, int tag)
+{
+  patch_beta_t *patch;
+  patch_t *dest;
+  int i, width, num_post;
+  size_t size, delta_header;
+
+  const size_t padding = 2 * sizeof(char);
+
+  if (strncasecmp("DOOMPRES", W_WadNameForLump(num), 8))
+    return (patch_t *)W_CacheLumpNum(num, tag);
+
+  patch = W_CacheLumpNum(num, tag);
+  size = W_LumpLength(num);
+  width = SHORT(patch->width);
+
+  num_post = 0;
+
+  for (i = 0; i < width; ++i)
+  {
+    int offset = (uint16_t)SHORT(patch->columnofs[i]);
+    byte *rover = (byte *)patch + offset;
+    while (*rover != 0xff)
+    {
+      post_t *column = (post_t *)rover;
+      rover += sizeof(column_t) + column->length;
+      ++num_post;
+    }
+  }
+
+  delta_header = width * (sizeof(int) - sizeof(short));
+
+  dest = Z_Calloc(1, size + delta_header + num_post * padding, PU_STATIC, 0);
```

Comment:
We need to cache this, or else multiple calls for the same patch lump will leak memory.

Reply:
Yes, I thought about how to organize it. Are the current `R_PatchByNum` functions is ok?

Reply:
Yes, looks clean, but we should probably make standard post-beta patches the default case.

Reply:
> Also, how about function pointers?

Ok, I will try it.